### PR TITLE
synchronize override methods

### PR DESCRIPTION
**Pull request by joshhale**
* Synchronize method that override a synchronized method

These were identified as warnings for [this project on lgtm.com](https://lgtm.com/projects/g/jenkinsci/jenkins/alerts/?mode=list). I didn't fix one or two of the warnings because the method implementation was too trivial to need any synchronization.

### CODE REVIEW

File: core/src/main/java/hudson/ExtensionListView.java [hunk 1]
```diff
@@ -107,12 +107,12 @@ public boolean remove(Object o) {
             }
 
             @Override
-            public void add(T t) {
+            synchronized public void add(T t) {
```
Comment (jvz) [inline core/src/main/java/hudson/ExtensionListView.java]: Typical ordering is `public synchronized void`

Comment (joshhale) [reply core/src/main/java/hudson/ExtensionListView.java]: ok, I'll switch the order

Comment (joshhale) [reply core/src/main/java/hudson/ExtensionListView.java]: I did a `sed -i` and also flipped one existing case of `synchronized public` (there are now 225 cases of `public synchronized`, so you are certainly right that it is typical!).

**Review by jvz — APPROVED** [review]

**Review by varyvol — APPROVED** [review]

**Comment by Wadeck** [comment]
Due to the number of synchronized added here, I would like to have more people reviewing the PR. It seems fine at first glance but I prefer to be sure it will not introduce any deadlock.

**Comment by alecharp** [comment]
I understand the reason why we would like to have the keyword `synchronized` on the methods, but also, to me, it's dangerous. Do we have tests covering this behavior? Is there a bug / performance issue this is solving (if so, we should have tests). 

It's not really like adding a `@NonNull` annotation (&copy; @varyvol :wink:)

**Comment by jvz** [comment]
The most ideal way I've found for testing synchronized access is to rely on SpotBugs' own static analysis. Make sure to use `@GuardedBy` on the relevant fields that require synchronization, and then SpotBugs can detect any uses that could cause issues. I've found that it even detected highly improbable issues that were nice to fix before they got abused somehow.

**Comment by alecharp** [comment]
@joshhale would you have time to add the `@GuardedBy` like @jvz suggested? This would help to see if spotbugs has anything to say.. Thanks.

**Comment by varyvol** [comment]
We are proposing this closure given there has been no follow-up for some time now.

**Comment by joshhale** [comment]
Sorry, I thought I already closed this. :bowing_man:

**Pull request closed without merging**